js/latoken.js

```diff
@@ -0,0 +1,658 @@
+            };
+            let limits = {
+                'amount': {
+                    'min': this.safeFloat (market, 'minQty'),
+                    'max': undefined,
+                },
+                'price': {
+                    'min': Math.pow (10, -precision['price']),
+                    'max': undefined,
+                },
+            };
+            limits['cost'] = {
+                'min': market['amountPrecision'],
+                'max': undefined,
+            };
+            result.push ({
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'active': true,
+                'precision': precision,
+                'limits': limits,
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    calculateFee (symbol, side, amount, price, takerOrMaker = 'taker') {
+        let market = this.markets[symbol];
+        let key = 'quote';
+        let rate = market[takerOrMaker];
+        let cost = amount * rate;
+        let precision = market['precision']['price'];
+        if (side === 'sell') {
+            cost *= price;
+        } else {
+            key = 'base';
+            precision = market['precision']['amount'];
+        }
+        cost = this.decimalToPrecision (cost, ROUND, precision, this.precisionMode);
+        return {
+            'type': takerOrMaker,
+            'currency': market[key],
+            'rate': rate,
+            'cost': parseFloat (cost),
+        };
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        let response = await this.privateGetAccountBalances (params);
+        let result = {
+            'info': response,
+        };
+        for (let i = 0; i < response.length; i++) {
+            let balance = response[i];
+            let currency = balance['symbol'];
+            let account = {
+                'free': parseFloat (balance['available']),
+                'used': parseFloat (balance['frozen']),
+                'total': parseFloat (balance['amount']),
+            };
+            result[currency] = account;
+        }
+        return this.parseBalance (result);
+    }
+
+    parseBidAsk (bidask) {
+        let price = parseFloat (bidask['price']);
+        let amount = parseFloat (bidask['amount']);
+        return {
+            'price': price,
+            'amount': amount,
+        };
+    }
+
+    parseBidsAsks (bidasks) {
+        let res = [];
+        for (let i = 0; i < bidasks.length; i++) {
+            res.push (this.parseBidAsk (bidasks[i]));
+        }
+        return res;
+    }
+
+    parseOrderBook (orderbook, timestamp = undefined, asksKey = 'asks', bidsKey = 'bids') {
+        return {
+            'bids': (bidsKey in orderbook) ? this.parseBidsAsks (orderbook[bidsKey]) : null,
```

Comment:
Python doesn't not recognize `null` here, it wants `undefined` (which is transpiled to `None`). The `null` value is JS-only. Just change it to `undefined` here.

Actually, I suggest to remove this `parseOrderBook` function entirely as well as `parseBidsAsks`, and use the built-in base methods for that. In other words, those methods are already implemented in a generic way in the base class, so we can just call them from the base class, no need to copy-paste them in the derived class.

Reply:
with base methods i have NaN in bids and asks

Reply:
@PolozovT can you post the raw unparsed exchange response here?

Add `this.verbose = true` to the `fetchOrderBook` before sending the request to see the response as is.

Surround it with triple backticks for readability and paste here:

&#096;&#096;&#096;
output
&#096;&#096;&#096;

We'll tell how to handle it with base methods.

Reply:
{"pairId":29,"symbol":"LAETH","spread":100.0,"asks":[{"price":53181181.0,"amount":0.01213456}],"bids":[]}

Reply:
ANSWER {
  bids: [],
  asks: [ [ NaN, NaN ] ],
  timestamp: 1560428064393,
  datetime: '2019-06-13T12:14:24.393Z',
  nonce: 1560428064877
}

Reply:
```JavaScript
// const response = await this.publicGetMarketDataOrderBook (this.extend (request, params));
const response = {"pairId":29,"symbol":"LAETH","spread":100.0,"asks":[{"price":53181181.0,"amount":0.01213456}],"bids":[]}
return this.parseOrderBook (response, undefined, 'bids', 'asks', 'price', 'amount');
```

Reply:
Tested it on my side with cli and it works:

```
mbp:ccxt igorkroitor$ node examples/js/cli binance parseOrderBook '{"pairId":29,"symbol":"LAETH","spread":100.0,"asks":[{"price":53181181.0,"amount":0.01213456}],"bids":[]}' undefined bids asks price amount
binance.parseOrderBook ([object Object], , bids, asks, price, amount)
{      bids: [],
       asks: [[53181181, 0.01213456]],
  timestamp:    undefined,
   datetime:    undefined,
      nonce:    undefined              }
mbp:ccxt igorkroitor$
```

Reply:
done) thank!

Reply:
Let us know if you have further questions, I suggest to not open a new issue for each new one, but just post here among the lines, we could then keep it within the context. Thx to you too!